plugin/evm/vm.go

```diff
@@ -838,6 +844,26 @@ func (vm *VM) initBlockBuilding() error {
 	return nil
 }
 
+func (vm *VM) WaitForEvent(ctx context.Context) (commonEng.Message, error) {
+	vm.builderLock.Lock()
+	builder := vm.builder
+	vm.builderLock.Unlock()
+
+	// Block building is not initialized yet, so we haven't finished syncing or bootstrapping.
+	if builder == nil {
+		select {
+		case <-ctx.Done():
+			return 0, ctx.Err()
+		case <-vm.stateSyncDone:
+			return commonEng.StateSyncDone, nil
+		case <-vm.shutdownChan:
+			return commonEng.Message(0), errShuttingDownVM
+		}
+	}
+
+	return vm.builder.waitForEvent(ctx)
```

Comment:
While this is not required, I feel like this makes a hair more sense.
```suggestion
	return builder.waitForEvent(ctx)
```